move/interchain_token_service/sources/interchain_token_service.move

```diff
@@ -118,13 +118,18 @@ module interchain_token_service::interchain_token_service {
     // ----------------
     // Public Functions
     // ----------------
+
+    /// Legacy function to register a coin from the given `CoinInfo`.
+    /// @deprecated since version 1.0
     #[allow(dead_code)]
     public fun register_coin<T>(self: &mut InterchainTokenService, coin_info: CoinInfo<T>, coin_management: CoinManagement<T>): TokenId {
         abort EUnsupported;
         let value = self.value_mut!(b"register_coin");
-        value.register_coin(coin_info, coin_management)
+        value.register_coin(coin_info, coin_management, option::is_some(coin_info.metadata()))
     }
 
+    /// Register a coin from user supplied values. Replaces legacy function `register_coin`.
+    /// @since version 1.0
```

Comment:
```suggestion
```
don't think we need to add comments for this, we should rely on version control instead. of npm package version is different from the internal version tag in the contract, so the label here can be confusing